libs/runtime/ebpf_work_queue.c

```diff
@@ -50,7 +50,19 @@ ebpf_timed_work_queue_create(
 
     KeInitializeTimer(&local_work_queue->timer);
     KeInitializeDpc(&local_work_queue->dpc, _ebpf_timed_work_queue_timer_callback, local_work_queue);
-    KeSetTargetProcessorDpc(&local_work_queue->dpc, (CCHAR)cpu_id);
+
+    PROCESSOR_NUMBER processor_number;
+    NTSTATUS status = KeGetProcessorNumberFromIndex(cpu_id, &processor_number);
+    if (!NT_SUCCESS(status)) {
+        return_value = EBPF_INVALID_ARGUMENT;
+        goto Done;
+    }
+
+    status = KeSetTargetProcessorDpcEx(&local_work_queue->dpc, &processor_number);
+    if (!NT_SUCCESS(status)) {
+        return_value = EBPF_INVALID_ARGUMENT;
```

Comment:
This is fine for this PR since it's consistent with the rest of the code.
However, it makes me wonder if ebpf_error.c (which has `ebpf_result_to_ntstatus()`) should expose another function to map in the opposite direction.